x/curating/handler.go

```diff
@@ -48,7 +48,17 @@ func handleMsgPost(ctx sdk.Context, k Keeper, msg types.MsgPost) (*sdk.Result, e
 func handleMsgUpvote(ctx sdk.Context, k Keeper, msg types.MsgUpvote) (*sdk.Result, error) {
 	deposit := k.GetParams(ctx).UpvoteDeposit
 
-	err := k.CreateUpvote(
+	_, found, err := k.GetUpvote(ctx, msg.VendorID, msg.PostID, msg.Curator)
```

Comment:
I think it would be better if this is added directly in the CreateUpvote method, so other methods that might call this in the future are enforced by this constraint.

Reply:
Agreed.